src/v/kafka/server/group_manager.cc

```diff
@@ -282,6 +282,7 @@ ss::future<size_t> group_manager::delete_offsets(
     if (group->in_state(group_state::dead)) {
         auto it = _groups.find(group->id());
         if (it != _groups.end() && it->second == group) {
+            co_await it->second->shutdown();
```

Comment:
i've never looked at what this group gate is protecting. by shutting down here do we risk any sort of hang/deadlock? are there other things to do / conditions to meet, before shutting down?

Reply:
I think we are good here, the code path in `delete_partition` does essentially the same thing